Stabilize live collections (#14550)
withastro/astro: .changeset/stale-ads-see.md

---
'astro': minor
---

Adds support for live content collections

Live content collections are a new type of [content collection](https://docs.astro.build/en/guides/content-collections/) that fetch their data at runtime rather than build time. This allows you to access frequently-updated data from CMSs, APIs, databases, or other sources using a unified API, without needing to rebuild your site when the data changes.

#### Live collections vs build-time collections

In Astro 5.0, the content layer API added support for adding diverse content sources to content collections. You can create loaders that fetch data from any source at build time, and then access it inside a page via `getEntry()` and `getCollection()`. The data is cached between builds, giving fast access and updates. 

> However there is no method for updating the data store between builds, meaning any updates to the data need a full site deploy, even if the pages are rendered on-demand. This means that content collections are not suitable for pages that update frequently. Instead, today these pages tend to access the APIs directly in the frontmatter. This works, but leads to a lot of boilerplate, and means users don't benefit from the simple, unified API that content loaders offer. In most cases users tend to individually create loader libraries that they share between pages.

Reviewer: ```suggestion
However there was no method for updating the data store between builds, meaning any updates to the data need a full site deploy, even if the pages are rendered on-demand. This meant that content collections were not suitable for pages that update frequently. Instead, these pages tended to access the APIs directly in the frontmatter. This worked, but leads to a lot of boilerplate, and meant users don't benefit from the simple, unified API that content loaders offer. In most cases users tended to individually create loader libraries that they share between pages.
```
This is now past tense, talking about Astro 5!
